pkg/expression/scalar_function.go

```diff
@@ -564,8 +577,21 @@ func (sf *ScalarFunction) EvalVectorFloat32(ctx EvalContext, row chunk.Row) (typ
 // HashCode implements Expression interface.
 func (sf *ScalarFunction) HashCode() []byte {
 	if len(sf.hashcode) > 0 {
+		if intest.InTest {
+			copyhashcode := make([]byte, len(sf.hashcode))
```

Comment:
and i didn't get what problem we are gonna solving. 
i think the main problem is Rehashing when the expression is changed?

Reply:
Yes, so we need to find who update the expression and forget to rehash.